jina/executors/indexers/keyvalue.py

```diff
@@ -66,7 +68,7 @@ def add(self, keys: Iterator[int], values: Iterator[bytes], *args, **kwargs):
             self._start += l
             self.write_handler.body.write(value)
             self._size += 1
-            # print(f'l: {l} p: {p} r: {r} r+l: {r + l} size: {self._size}')
+        self.write_handler.flush()
```

Comment:
shoul we maybe have a lazy flush? and flush only if we need the guarantee? or it feels like early optimization?

Reply:
Opened an issue https://github.com/jina-ai/jina/issues/1413